### PR TITLE
CEM-2130 add search functionality to ListHeader component in react-ui

### PR DESCRIPTION
**Pull request by roggc**
lhwithsb

### CODE REVIEW

**Comment by Ampatte2** [comment]
Video of how this works at 255px width?

File: src/Containers/List/ListHeader.tsx [hunk 1]
```diff
@@ -65,4 +103,15 @@ const Row = styled.div<RowProps>`
...
     ${(props): string | undefined => props.iconProps}
+    cursor:pointer;
 `;
+
+const IconsContainer=styled.div<RowProps>`
+${(props): string | undefined =>
+        props.display && Mixins.flex(props.display)};
+`
+
+const SearchInput=styled(Input)`
```
Comment (ralph-dev) [inline src/Containers/List/ListHeader.tsx:114]: use the Existing React-UI Input

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Expansion Effect is incorrect (It should expand from the Search Icon)
Try Placing a Grey Border around the Icon and Expanding that to get the effect :)

File: src/Containers/List/ListHeader.tsx (hunk 1 above)
Comment (roggc) [reply src/Containers/List/ListHeader.tsx:114]: Yes, it is the existing React-ui input

**Comment by Ampatte2** [comment]
@ralph-dev this style of search is going to be an issue with this component and it's current use cases. It's rarely at fullscreen width. Terminal and dash both use a 255px width for the list.

Possibly make the search drop down below at certain widths?, or transition the search field beneath the header?

**Comment by ralph-dev** [comment]
> @ralph-dev this style of search is going to be an issue with this component and it's current use cases. It's rarely at fullscreen width. Terminal and dash both use a 255px width for the list.
> 
> Possibly make the search drop down below at certain widths?, or transition the search field beneath the header?

This only searches for Order ID, what would be the issue if it takes up the full width of 255 px?

**Comment by Ampatte2** [comment]
It can take up the full 255, right now the search input is inline with the header so it'll look like the small mobile viewport in the second pic.

File: src/Containers/List/ListHeader.tsx [hunk 2]
```diff
@@ -25,31 +29,55 @@ export const ListHeader: React.FC<ListHeaderProps> = ({
...
-            {icon && (
-                <Icon as={icon} onClick={iconClick} iconProps={iconProps} />
-            )}
-            {headerRowComponent}
-        </Row>
-        {children}
-    </Header>
-);
+}): React.ReactElement => {
+    const stateExpanded=useState(false)
```
Comment (ralph-dev) [inline src/Containers/List/ListHeader.tsx:35]: Don't use this , write out both set and the original

**Review by ralph-dev — APPROVED** [review]
Lgtm

**Pull request merged**